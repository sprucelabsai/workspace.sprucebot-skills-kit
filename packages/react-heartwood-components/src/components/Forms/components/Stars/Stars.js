// @flow
import React from 'react'
import StarIcon from '../../../../../static/assets/icons/Social-Medias-Rewards-Rating/Rating/rating-star-1.svg'

const Stars = () => {
	return (
		<div className="stars">
			<label className="star-input__wrapper">
				<input className="star-input" type="radio" name="stars" value="1" />
				<StarIcon className="star" />
			</label>
			<label className="star-input__wrapper">
				<input className="star-input" type="radio" name="stars" value="2" />
				<StarIcon className="star" />
				<StarIcon className="star" />
			</label>
			<label className="star-input__wrapper">
				<input className="star-input" type="radio" name="stars" value="3" />
				<StarIcon className="star" />
				<StarIcon className="star" />
				<StarIcon className="star" />
			</label>
			<label className="star-input__wrapper">
				<input className="star-input" type="radio" name="stars" value="4" />
				<StarIcon className="star" />
				<StarIcon className="star" />
				<StarIcon className="star" />
				<StarIcon className="star" />
			</label>
			<label className="star-input__wrapper">
				<input className="star-input" type="radio" name="stars" value="5" />
				<StarIcon className="star" />
				<StarIcon className="star" />
				<StarIcon className="star" />
				<StarIcon className="star" />
				<StarIcon className="star" />
			</label>
		</div>
	)
}

export default Stars
