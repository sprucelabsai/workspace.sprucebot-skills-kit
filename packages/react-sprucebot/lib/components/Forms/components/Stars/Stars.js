"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var StarIcon = function StarIcon(props) {
  return _react.default.createElement("svg", props, _react.default.createElement("path", {
    clipRule: "evenodd",
    d: "M11.551 5.745a.5.5 0 0 1 .9 0L14.3 9.483l4.144.6a.5.5 0 0 1 .276.854l-3 2.9.708 4.1a.5.5 0 0 1-.724.528L12 16.531 8.289 18.47a.5.5 0 0 1-.724-.528l.708-4.1-3-2.9a.5.5 0 0 1 .276-.854l4.145-.6 1.857-3.743z",
    strokeLinecap: "round",
    strokeLinejoin: "round"
  }));
};

StarIcon.defaultProps = {
  width: "24",
  height: "24",
  viewBox: "0 0 24 24",
  xmlns: "http://www.w3.org/2000/svg"
};

var Stars = function Stars(props) {
  return _react.default.createElement("div", {
    className: "stars"
  }, _react.default.createElement("label", {
    className: "star-input__wrapper"
  }, _react.default.createElement("input", {
    className: "star-input",
    type: "radio",
    name: "stars",
    value: "1"
  }), _react.default.createElement(StarIcon, {
    className: "star"
  })), _react.default.createElement("label", {
    className: "star-input__wrapper"
  }, _react.default.createElement("input", {
    className: "star-input",
    type: "radio",
    name: "stars",
    value: "2"
  }), _react.default.createElement(StarIcon, {
    className: "star"
  }), _react.default.createElement(StarIcon, {
    className: "star"
  })), _react.default.createElement("label", {
    className: "star-input__wrapper"
  }, _react.default.createElement("input", {
    className: "star-input",
    type: "radio",
    name: "stars",
    value: "3"
  }), _react.default.createElement(StarIcon, {
    className: "star"
  }), _react.default.createElement(StarIcon, {
    className: "star"
  }), _react.default.createElement(StarIcon, {
    className: "star"
  })), _react.default.createElement("label", {
    className: "star-input__wrapper"
  }, _react.default.createElement("input", {
    className: "star-input",
    type: "radio",
    name: "stars",
    value: "4"
  }), _react.default.createElement(StarIcon, {
    className: "star"
  }), _react.default.createElement(StarIcon, {
    className: "star"
  }), _react.default.createElement(StarIcon, {
    className: "star"
  }), _react.default.createElement(StarIcon, {
    className: "star"
  })), _react.default.createElement("label", {
    className: "star-input__wrapper"
  }, _react.default.createElement("input", {
    className: "star-input",
    type: "radio",
    name: "stars",
    value: "5"
  }), _react.default.createElement(StarIcon, {
    className: "star"
  }), _react.default.createElement(StarIcon, {
    className: "star"
  }), _react.default.createElement(StarIcon, {
    className: "star"
  }), _react.default.createElement(StarIcon, {
    className: "star"
  }), _react.default.createElement(StarIcon, {
    className: "star"
  })));
};

var _default = Stars;
exports.default = _default;