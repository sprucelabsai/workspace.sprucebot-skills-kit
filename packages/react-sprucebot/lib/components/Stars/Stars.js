'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Stars = function (_Component) {
	_inherits(Stars, _Component);

	function Stars(props) {
		_classCallCheck(this, Stars);

		var _this = _possibleConstructorReturn(this, (Stars.__proto__ || Object.getPrototypeOf(Stars)).call(this, props));

		_this.state = {
			score: props.score,
			hover: 0
		};
		return _this;
	}

	_createClass(Stars, [{
		key: 'onClickStar',
		value: function onClickStar(score, e) {
			var _this2 = this;

			if (!this.props.static) {
				this.setState(function (prevState) {
					if (prevState.score !== score) {
						if (_this2.props.onChange) {
							_this2.props.onChange(score, e);
						}
						return {
							score: score
						};
					}

					return {};
				});
			}
		}
	}, {
		key: 'onMouseOverStar',
		value: function onMouseOverStar(score, e) {
			if (!this.props.static) {
				this.setState({
					hover: score
				});
			}
		}
	}, {
		key: 'onMouseLeave',
		value: function onMouseLeave(e) {
			if (!this.props.static) {
				this.setState({
					hover: 0
				});
			}
		}
	}, {
		key: 'render',
		value: function render() {
			var _this3 = this;

			var _props = this.props,
			    max = _props.max,
			    onChange = _props.onChange;
			var _state = this.state,
			    score = _state.score,
			    hover = _state.hover;

			// just round for score until halves are possible

			score = Math.round(score);

			if (hover > 0) {
				score = hover;
			}

			var stars = [];

			var _loop = function _loop(idx) {
				stars.push(_react2.default.createElement('div', {
					onMouseOver: function onMouseOver(e) {
						_this3.onMouseOverStar(idx, e);
					},
					onClick: function onClick(e) {
						_this3.onClickStar(idx, e);
					},
					key: 'star-' + idx,
					className: 'star ' + (score >= idx ? 'active' : '')
				}));
			};

			for (var idx = 1; idx <= max; idx++) {
				_loop(idx);
			}

			return _react2.default.createElement(
				'div',
				{
					className: 'stars',
					onMouseLeave: function onMouseLeave(e) {
						_this3.onMouseLeave(e);
					}
				},
				stars
			);
		}
	}]);

	return Stars;
}(_react.Component);

exports.default = Stars;


Stars.propTypes = {
	score: _propTypes2.default.number,
	max: _propTypes2.default.number,
	onChange: _propTypes2.default.func,
	static: _propTypes2.default.bool
};

Stars.defaultProps = {
	max: 5,
	score: 0,
	static: false,
	onChange: function onChange(score, e) {}
};