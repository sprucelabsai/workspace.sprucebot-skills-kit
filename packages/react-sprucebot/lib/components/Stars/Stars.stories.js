'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _react3 = require('@storybook/react');

var _addonInfo = require('@storybook/addon-info');

var _react4 = require('@storybook/addon-knobs/react');

var _storybookReadme = require('storybook-readme');

var _Stars = require('./Stars');

var _Stars2 = _interopRequireDefault(_Stars);

var _Stars3 = require('./Stars.md');

var _Stars4 = _interopRequireDefault(_Stars3);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var stories = (0, _react3.storiesOf)('Stars', module);
stories.addDecorator(_react4.withKnobs);

stories.add('Interactive', (0, _storybookReadme.withReadme)(_Stars4.default, function () {
  return _react2.default.createElement(_Stars2.default, null);
}));