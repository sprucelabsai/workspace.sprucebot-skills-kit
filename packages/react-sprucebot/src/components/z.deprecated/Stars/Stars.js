import React, { Component } from 'react'
import PropTypes from 'prop-types'

export default class Stars extends Component {
	constructor(props) {
		super(props)
		this.state = {
			score: props.score,
			hover: 0
		}
	}
	onClickStar(score, e) {
		if (!this.props.static) {
			this.setState(prevState => {
				if (prevState.score !== score) {
					if (this.props.onChange) {
						this.props.onChange(score, e)
					}
					return {
						score
					}
				}

				return {}
			})
		}
	}
	onMouseOverStar(score, e) {
		if (!this.props.static) {
			this.setState({
				hover: score
			})
		}
	}
	onMouseLeave(e) {
		if (!this.props.static) {
			this.setState({
				hover: 0
			})
		}
	}
	render() {
		const { max, onChange } = this.props
		let { score, hover } = this.state

		// just round for score until halves are possible
		score = Math.round(score)

		if (hover > 0) {
			score = hover
		}

		const stars = []
		for (let idx = 1; idx <= max; idx++) {
			stars.push(
				<div
					onMouseOver={e => {
						this.onMouseOverStar(idx, e)
					}}
					onClick={e => {
						this.onClickStar(idx, e)
					}}
					key={`star-${idx}`}
					className={`star ${score >= idx ? 'active' : ''}`}
				/>
			)
		}

		return (
			<div
				className="stars"
				onMouseLeave={e => {
					this.onMouseLeave(e)
				}}
			>
				{stars}
			</div>
		)
	}
}

Stars.propTypes = {
	score: PropTypes.number,
	max: PropTypes.number,
	onChange: PropTypes.func,
	static: PropTypes.bool
}

Stars.defaultProps = {
	max: 5,
	score: 0,
	static: false,
	onChange: (score, e) => {}
}
