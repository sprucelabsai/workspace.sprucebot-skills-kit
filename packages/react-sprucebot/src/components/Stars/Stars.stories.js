import React from 'react'
import { storiesOf } from '@storybook/react'
import { withInfo } from '@storybook/addon-info'
import { withKnobs } from '@storybook/addon-knobs/react'
import { withReadme } from 'storybook-readme'
import Stars from './Stars'
import readme from './Stars.md'

const stories = storiesOf('Stars', module)
stories.addDecorator(withKnobs)

stories.add('Interactive', withReadme(readme, () => <Stars />))
