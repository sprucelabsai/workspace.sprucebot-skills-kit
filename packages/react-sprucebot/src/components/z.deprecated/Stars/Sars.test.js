import React from 'react'
import Stars from './Stars'
import { shallow } from 'enzyme'

describe('Stars tests', () => {
	it('Should match the snapshot', () => {
		const renderedComponent = shallow(<Stars />)

		expect(renderedComponent).toMatchSnapshot()
	})
})
