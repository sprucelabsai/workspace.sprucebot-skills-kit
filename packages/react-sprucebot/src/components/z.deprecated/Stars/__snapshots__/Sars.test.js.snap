// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stars tests Should match the snapshot 1`] = `
<div
  className="stars"
  onMouseLeave={[Function]}
>
  <div
    className="star "
    key="star-1"
    onClick={[Function]}
    onMouseOver={[Function]}
  />
  <div
    className="star "
    key="star-2"
    onClick={[Function]}
    onMouseOver={[Function]}
  />
  <div
    className="star "
    key="star-3"
    onClick={[Function]}
    onMouseOver={[Function]}
  />
  <div
    className="star "
    key="star-4"
    onClick={[Function]}
    onMouseOver={[Function]}
  />
  <div
    className="star "
    key="star-5"
    onClick={[Function]}
    onMouseOver={[Function]}
  />
</div>
`;
